pkg/resources/schema.go

```diff
@@ -52,6 +58,50 @@ var schemaSchema = map[string]*schema.Schema{
 	},
 }
 
+type schemaID struct {
+	DatabaseName string
+	SchemaName   string
+}
+
+// String() takes in a schemaID object and returns a pipe-delimited string:
+// DatabaseName|schemaName
+func (si *schemaID) String() (string, error) {
+	var buf bytes.Buffer
+	csvWriter := csv.NewWriter(&buf)
+	csvWriter.Comma = schemaIDDelimiter
+	dataIdentifiers := [][]string{{si.DatabaseName, si.SchemaName}}
+	err := csvWriter.WriteAll(dataIdentifiers)
+	if err != nil {
+		return "", err
+	}
+	strSchemaID := strings.TrimSpace(buf.String())
+	return strSchemaID, nil
+}
+
+// schemaIDFromString() takes in a pipe-delimited string: DatabaseName|schemaName
+// and returns a schemaID object
+func schemaIDFromString(stringID string) (*schemaID, error) {
```

Comment:
also needs tests

Reply:
oh huh... I wrote schema_internal_test.go
not sure why it isn't being run